src/main/java/makima/command/Makima.java

```diff
@@ -0,0 +1,235 @@
+    public static final Scanner SC = new Scanner(System.in);
+    private boolean isRunning = true;
+    private boolean editedTasks;
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+
+    private void greet() {
+        System.out.println("Yahallo! I'm your friendly chatbot, makima.Makima!");
+        System.out.println("What can I do for you? *_*");
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    private void farewell() {
+        System.out.println("Baibai. Hope to see you soon! ^_^");
+        System.out.println(LINE_SEPERATOR);
+    }
+
+    private void displayList() {
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println(i+1 + ":" + tasks.get(i));
+        }
+    }
+
+    /**
+     * Prints a prompt and returns a valid list index.
+     *
+     * @param prompt
+     * @return list index
+     */
+    private int getListIndex(String prompt) {
+        System.out.println(prompt);
+        System.out.println(LINE_SEPERATOR);
+        return getListIndex();
+    }
+
+    /**
+     * Returns a 1-indexed list index.
+     * Repeatedly scans user input until a valid index is written.
+     *
+     * @return list index
+     */
+    private int getListIndex() {
+        while (true) {
+            String userInput = Parser.getInput();
+            try {
+                int index = Integer.parseInt(userInput);
+                if (index < 1 || index > tasks.size()) {
+                    System.out.println("Invalid index");
+                } else {
+                    return index-1;
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Please enter a number!");
+            }
+        }
+    }
+
+    /**
+     * Returns a non-empty user input.
+     *
+     * @param prompt
+     * @return user input
+     */
+    private String getInput(String prompt) {
+        System.out.println(prompt);
+        System.out.println(LINE_SEPERATOR);
+        return Parser.getInput();
+    }
+
+    /**
+     * Returns a local date time.
+     * Repeatedly scans user input until a valid date time is written.
+     * Date time is in the format YYYY-MM-DDTHH:MM
+     *
+     * @param prompt
+     * @return date time
+     */
+    private LocalDateTime getDate(String prompt) {
+        System.out.println(prompt);
+        System.out.println(LINE_SEPERATOR);
+        return Parser.getDate();
+    }
+
+    /**
+     * Returns a local date time that is AFTER the specified date.
+     *
+     * @param date
+     * @param prompt
+     * @return date time
+     */
+    private LocalDateTime getDateAfter(LocalDateTime date, String prompt) {
+        System.out.println(prompt);
+        System.out.println(LINE_SEPERATOR);
+        return Parser.getDateAfter(date);
+    }
+
+    /**
+     * Prints a confirmation for completion of commands which
+     * edit the tasks array.
+     */
+    private void done() {

```

Comment:
Is it possible to name this method with a word / phrase that is a more distinctive verb?